File: AuctionMarket/src/main/java/com/portfolio/auctionmarket/domain/orders/dto/OrderResponse.java
package com.portfolio.auctionmarket.domain.orders.dto;

import com.fasterxml.jackson.annotation.JsonFormat;
import com.fasterxml.jackson.annotation.JsonUnwrapped;
import com.portfolio.auctionmarket.domain.orders.entity.Order;
import com.portfolio.auctionmarket.domain.products.dto.ProductResponse;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Getter
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class OrderResponse {
    private Long orderId;
    private String nickname;
    private Long finalPrice;

    @JsonUnwrapped
    private ProductResponse productResponse;

    @JsonFormat(pattern = "yyyy-MM-dd'T'HH:mm:ss")
    private LocalDateTime endTime;

    public static OrderResponse from(Order entity) {
        if (entity == null) {
            return null;
        }
        return OrderResponse.builder()
                .orderId(entity.getOrderId())
                .nickname(entity.getBuyer().getNickname())
                .finalPrice(entity.getFinalPrice())
                .endTime(entity.getAuction().getEndTime())
                .productResponse(ProductResponse.from(entity.getAuction().getProduct()))
                .build();
    }
}

File: AuctionMarket/auction-app-user/src/main/kotlin/org/example/auctioncommon/domain/orders/dto/OrderResponse.kt
package org.example.auctioncommon.domain.orders.dto

import com.fasterxml.jackson.annotation.JsonFormat
import com.fasterxml.jackson.annotation.JsonUnwrapped
import org.example.auctioncommon.domain.order.entity.Order
import org.example.auctioncommon.domain.product.dto.ProductResponse
import org.example.auctioncommon.domain.product.dto.toProductResponse

data class OrderResponse(
    val orderId: Long,
    val nickname: String,
    val finalPrice: Long,

    @field:JsonUnwrapped
    val productResponse: ProductResponse,

    @field:JsonFormat(pattern = "yyyy-MM-dd'T'HH:mm:ss")
    val endTime: java.time.LocalDateTime? = null,
)
fun Order.toDto() : OrderResponse {
    return OrderResponse(
        orderId = this.orderId ?: 0L,
        nickname = this.buyer?.nickname.toString(),
        finalPrice = this.finalPrice ?: 0L,
        endTime = this.auction.endTime,
        productResponse = this.auction.product?.toProductResponse()!!
    )
}

